subset/bacnet/bacnetTests/src/main/java/FauxDeviceEngine/EntryPoint.java

```diff
@@ -0,0 +1,129 @@
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+public class EntryPoint {
+
+    private static int deviceId = 0;
+    private static IpNetwork network;
+    private static LocalDevice localDevice;
+    private static String fauxDeviceJSONFilename = "";
+    private static int timeout = 1000;
+
+    public static void main(String[] args) {
+        if (args.length != 3) {
+            throw new RuntimeException("Usage: localIpAddr broadcastIpAddr fauxDeviceJSONFilename");
+        }
+        String localIpAddr = args[0];
+        String broadcastIpAddr = args[1];
+        fauxDeviceJSONFilename = args[2];
+
+        int port = IpNetwork.DEFAULT_PORT;
+        network = new IpNetwork(broadcastIpAddr, port,
+                IpNetwork.DEFAULT_BIND_IP, 0, localIpAddr);
+        Transport transport = new Transport(network);
+        transport.setTimeout(timeout);
+
+        try {
+            JSONArray bacnetObjectArray = readJSONFile();
+            getDeviceID(bacnetObjectArray);
+            if(deviceId == 0) {
+                System.out.println("Device ID not found in JSON file. Generating random ID...");
+                deviceId = (int) Math.floor(Math.random() * 1000.0);
+            }
+            System.out.println("Creating LoopDevice id " + deviceId);
+            localDevice = new LocalDevice(deviceId, transport);
+            localDevice.getConfiguration().setProperty(PropertyIdentifier.modelName,
+                    new CharacterString("BACnet4J LoopDevice"));
+            System.out.println("Local device is running with device id " + deviceId);
+            addBacnetProperties(bacnetObjectArray);
+            localDevice.initialize();
+            System.out.println("Device initialized...");
+        } catch (RuntimeException e) {
+            System.out.println("Ex in LoopDevice() ");
+            e.printStackTrace();
+            localDevice.terminate();
+            localDevice = null;
+            throw e;
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    private static JSONArray readJSONFile() {
+        String jsonFile = fauxDeviceJSONFilename;
+        FileManager fileManager = new FileManager();
+        String absolute_path = fileManager.getAbsolutePath();
+        JSON json = new JSON(absolute_path + "tmp/" + jsonFile);
+        JSONArray bacnetObjectTypesList = json.read();
+        return bacnetObjectTypesList;
+    }
+
+    private static void addBacnetProperties(JSONArray bacnetObjectsList) {
+        bacnetObjectsList.forEach(bacnetObject -> addProperty((JSONObject) bacnetObject));
+    }
+
+    private static void getDeviceID(JSONArray bacnetObjectsList) {
+        bacnetObjectsList.forEach(bacnetObject -> getID((JSONObject) bacnetObject));
+    }
+
+    private static void getID(JSONObject bacnetObject) {
+        List<String> bacnetObjectTypeArr = new ArrayList<>(bacnetObject.keySet());
```

Comment:
I am trying to understand why the set is being converted to a list. Would it make sense if we used an iterator to get an (any?) element from the set?

Reply:
From what I can gather - this function is used solely by the faux device for use with integration testing. As such, the order of `bacnetObjectTypeArr` is always fixed (the device id always comes first - see [here](https://github.com/arupiot/daq/blob/bacnet_pics_test_discover/subset/bacnet/bacnetTests/src/main/resources/Faux-Device-Pass.json#L3) and [here](https://github.com/arupiot/daq/blob/bacnet_pics_test_discover/subset/bacnet/bacnetTests/src/main/resources/Faux-Device-Fail.json#L3))

I'd agree that order shouldn't matter, or at least that the `0` (`.get(0)`) index reference should be labelled, but this doesn't affect testing a device in the wild. That being said, I can certainly make the change as requested to iterate over the `ArrayList`?